Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h

```diff
@@ -1464,30 +1453,98 @@ bool remove_self_intersections_one_step(TriangleMesh& tm,
     }
     while(non_manifold_vertex_removed);
 
-  /// remove the selection
-    if (border_edges_found){
+    typedef CGAL::dynamic_face_property_t<std::size_t> Fid_tag;
+    typedef CGAL::Face_filtered_graph<TriangleMesh> Filtered_graph;
+    Filtered_graph filtered(tm, faces_to_remove);
```

Comment:
We should get rid of the `Filtered_graph` since we modify the mesh in the following while still using the face_graph.